lib/cadence/connection/thrift.rb

```diff
@@ -142,11 +148,12 @@ def poll_for_decision_task(domain:, task_list:)
         send_request('PollForDecisionTask', request)
       end
 
-      def respond_decision_task_completed(task_token:, decisions:)
+      def respond_decision_task_completed(task_token:, decisions:, query_results:)
         request = CadenceThrift::RespondDecisionTaskCompletedRequest.new(
           identity: identity,
           taskToken: task_token,
-          decisions: Array(decisions)
+          decisions: Array(decisions),
+          queryResults: query_results
```

Comment:
This should be `queryResults: query_results.transform_values { |value| Serializer.serialize(value)`